### DISCUSSION TITLE
Are negative time values ​​allowed in RAMP?

### DISCUSSION THREAD

**Discussion by mst-parkhachev** [firemodels/fds#13945] [General]
Dear colleagues!

Will the FDS behavior be the same for the following RAMPs?

1.
&RAMP ID='RAMP_1' T=0 F=0/
&RAMP ID='RAMP_1' T=1 F=1/

2.
&RAMP ID='RAMP_2' T=-1 F=-1/
&RAMP ID='RAMP_2' T=1 F=1/

Linear interpolation gives the same result for positive time values. However, are negative values of T ​​allowed in the FDS?

I understand that there is no need to use negative time values. However, I have a set of already generated files that use negative time values. In some cases, the time values ​​are not symmetrical around zero. For example, there are cases like this:

&RAMP ID='RAMP_3' T=-2 F=-1/
&RAMP ID='RAMP_3' T=-1 F=1/
&RAMP ID='RAMP_3' T=1 F=1/

&RAMP ID='RAMP_4' T=-1 F=-1/
&RAMP ID='RAMP_4' T=1 F=-1/
&RAMP ID='RAMP_4' T=2 F=1/

Can I use them? Will RAMP always give the correct result for positive times?

**Comment by drjfloyd** [comment]
A great way to become more proficient with using FDS is to get into the habit of making short little test cases to explore the effects of different inputs. For example with this case you could do something like a simple 10 x 10 x 10 cell box 1 m on a side with a burner on the floor. Run the case with RAMP 1 and then with RAMP 2 and see if you get the same or different results and see if you get errors for having negative time values.

Same with your other RAMPs. You could just try them and see if the result is what you expect or not.

**Reply by mst-parkhachev** [reply]
Let me assure you that I have a habit of spending a few minutes (оften a few weeks) researching a problem before posting a question here. Unfortunately, when proving anything there is a big difference between what is necessary and what is sufficient. While studying the results of some calculations, I had doubts that using negative time values ​​is always correct. Perhaps the problem was due to something else, but doubts had already arisen. Unfortunately, a few correctly working examples are not sufficient proof that this can always be done.

It seemed more reliable to me to check with the authors of the program. That is why I asked the question and gave several simple random examples as an illustration. As far as I understood your answer, taking into account my experiments, there are no problems with negative time values.

Just in case, I will note one observation. During the experiments, it was accidentally discovered that if RAMP_Q takes a negative value, then the value of MLR_FUEL in the file CHID_hrr.csv becomes negative.

**Reply by drjfloyd** [reply]
With a RAMP, FDS simply mulitpilies the input being adjusted by the interpolated F value. We do not prohibit negatve F values as RAMPs can be used for many things some of which a negative value is perfectly acceptable. 

There is no prohibition on having negative time in an FDS model. See T_BEGIN in the User's Guide.
